### PR TITLE
chore: warn user about full recirculation of fluids in a compressor s…

### PR DESCRIPTION
**Pull request by olelod**
…tage in a multiple streams and pressures compressor train

### CODE REVIEW

File: src/libecalc/core/models/compressor/train/variable_speed_compressor_train_common_shaft_multiple_streams_and_pressures.py [hunk 1]
```diff
@@ -872,6 +872,11 @@ def calculate_compressor_train_given_rate_ps_speed(
                     stage_number=stage_number
                 )
                 if fluid_to_recirculate:
+                    logger.warning(
+                        f"For stage number {stage_number}, there is no fluid entering the stage at at this time step. "
```
[the remark this answers is not in the document]
Comment (olelod) [reply src/libecalc/core/models/compressor/train/variable_speed_compressor_train_common_shaft_multiple_streams_and_pressures.py:876]: I agree, but I also think this is a more general problem that occurs many of the times we use `logger.warning`. This is at least not the only place. Should we address it in a more general setting?

[the remark this answers is not in the document]
Comment (TeeeJay) [reply src/libecalc/core/models/compressor/train/variable_speed_compressor_train_common_shaft_multiple_streams_and_pressures.py:876]: You are right...it is happening many times already, and we should instead find a better solution in a general setting. Later. (Unless you see and find a good solution for this and others now)

File: src/libecalc/core/models/compressor/results.py [hunk 2]
```diff
@@ -290,15 +290,21 @@ def set_failure_status(cls, v, values):
 
     @property
     def chart_area_status(self) -> ChartAreaFlag:
-        chart_area_failure = [
-            stage.chart_area_flag
-            for stage in self.stage_results
-            if stage.chart_area_flag != ChartAreaFlag.INTERNAL_POINT
-        ]
+        """Checks where the operational points are placed in relation to the compressor charts in a compressor train.
```
Comment (TeeeJay) [inline src/libecalc/core/models/compressor/results.py:293]: Set up docstring in PyCharm to use "Google" in "File > Settings > Tools > Python Integrated Tools > Docstring format" --> Google :)

File: src/libecalc/core/models/compressor/train/chart/variable_speed_compressor_chart.py [hunk 3]
```diff
@@ -317,24 +317,25 @@ def _evaluate_point_validity_chart_area_flag_and_adjusted_speed_and_rate(
...
-                    point_is_valid = False
             elif rate > maximum_flow_rate_for_speed:
                 chart_area_flag = (
                     ChartAreaFlag.BELOW_MINIMUM_SPEED_AND_ABOVE_MAXIMUM_FLOW_RATE
                     if speed_is_increased
                     else ChartAreaFlag.ABOVE_MAXIMUM_FLOW_RATE
                 )
                 point_is_valid = False
-
+            if increase_rate_left_of_minimum_flow_assuming_asv:
```
Comment (TeeeJay) [inline src/libecalc/core/models/compressor/train/chart/variable_speed_compressor_chart.py:335]: Is it on purpose that this moved outside the `elif`? One intendation out?

File: src/libecalc/core/models/compressor/train/chart/variable_speed_compressor_chart.py [hunk 4]
```diff
@@ -317,24 +317,25 @@ def _evaluate_point_validity_chart_area_flag_and_adjusted_speed_and_rate(
...
+                rate_to_use = max(minimum_flow_rate_for_speed, rate_to_use)
```
Comment (TeeeJay) [inline src/libecalc/core/models/compressor/train/chart/variable_speed_compressor_chart.py:336]: is `point_is_valid` = `False` here as well?

File: src/libecalc/core/models/compressor/train/chart/variable_speed_compressor_chart.py (hunk 3 above)
Comment (olelod) [reply src/libecalc/core/models/compressor/train/chart/variable_speed_compressor_chart.py]: Yes and no. Updating the rate to use can be seen as independent from finding the placement of the rate in the input relative to the compressor chart. Done some small updates now. Maybe more clear...?

File: src/libecalc/core/models/compressor/train/chart/variable_speed_compressor_chart.py (hunk 4 above)
Comment (olelod) [reply src/libecalc/core/models/compressor/train/chart/variable_speed_compressor_chart.py]: If `increase_rate_left_of_minimum_flow_assuming_asv == True `points will be move to minimum flow and considered to be valid.

File: src/libecalc/core/models/compressor/train/chart/variable_speed_compressor_chart.py [hunk 5]
```diff
@@ -317,16 +325,21 @@ def _evaluate_point_validity_chart_area_flag_and_adjusted_speed_and_rate(
             # Todo: Need to QA this logic. Below minimum speed requires speed < minimum_speed, but this is not enforced?
             minimum_flow_rate_for_speed = self.minimum_rate_as_function_of_speed(speed_to_use)
             maximum_flow_rate_for_speed = self.maximum_rate_as_function_of_speed(speed_to_use)
-            if rate < minimum_flow_rate_for_speed:
+            # first update rate_to_use when below minimum flow if increase_rate_left_of_minimum_flow_assuming_asv
+            if rate_to_use < minimum_flow_rate_for_speed:
+                if increase_rate_left_of_minimum_flow_assuming_asv:
+                    rate_to_use = max(minimum_flow_rate_for_speed, rate_to_use)
```
Comment (TeeeJay) [inline src/libecalc/core/models/compressor/train/chart/variable_speed_compressor_chart.py:331]: you already know that `rate_to_use < minimum_flow_rate_for_speed` ? Then just take `minimum_flow_rate_for_speed`?

**Review by TeeeJay — APPROVED** [review]
See comment. I think it is ok.

Do we have a test that catches these 2 scenarios somehow?

File: src/libecalc/core/models/compressor/train/chart/variable_speed_compressor_chart.py (hunk 5 above)
Comment (olelod) [reply src/libecalc/core/models/compressor/train/chart/variable_speed_compressor_chart.py:331]: 💡

**Comment by olelod** [comment]
> See comment. I think it is ok.
> 
> Do we have a test that catches these 2 scenarios somehow?

Made a test for full recirculation in an earlier PR. Added check of the ChartAreaFlag also in that test now.

**Comment by TeeeJay** [comment]
Remember to update changelog in docs as well (v8.5)

**Pull request merged**